tests/e2e/additional-containers-collector/00-assert-daemonset-without-additional-containers.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: apps/v1
```

Comment:
Any particular reason not to put all the assertions in one file for each step, like we do for other tests? This isn't a criticism per se, I'm honestly wondering if this makes the tests easier to maintain. WDYT @open-telemetry/operator-approvers ?

Reply:
Hm... I see the value in that, this certainly increases our surface area for testing. I do think we're generally due for some e2e test maintenance though so maybe we can do that in a follow up?

Reply:
We can discuss during the next SIG meeting. But I agree we shouldn't hold up this PR for this.